hstrat/phylogenetic_inference/pairwise/_calc_rank_of_mrca_bounds_between.py

```diff
@@ -128,13 +133,33 @@ def calc_rank_of_mrca_bounds_between(
             confidence_level_commonality=confidence_level,
             confidence_level_disparity=0.49,  # 0.49 i.e., definitive max
         )
-        assert (
-            first_disparity is not None
-            or first.GetNumStrataDeposited() == second.GetNumStrataDeposited()
-        )
+
+        # change in logic to handle surfaces potentially ignoring ranks
+        if isinstance(first, HereditaryStratigraphicColumn) and isinstance(
+            second, HereditaryStratigraphicColumn
+        ):
+            assert (
+                first_disparity is not None
+                or first.GetNumStrataDeposited()
+                == second.GetNumStrataDeposited()
+            )
+        if first_disparity is None:
+            print("in here")
+            max_rank_first = first.GetNumStrataDeposited() + (
+                first.S
+                if isinstance(first, HereditaryStratigraphicSurface)
+                else 0
+            )
+            max_rank_second = second.GetNumStrataDeposited() + (
+                second.S
+                if isinstance(second, HereditaryStratigraphicSurface)
+                else 0
```

Comment:
Isn't this changing behavior for the column? Previously it was `first.GetNumStrataDeposited()`, but now it's zero?